blinker/base.py

```diff
@@ -396,10 +396,7 @@ def signal(self, name, doc=None):
         Repeated calls to this function will return the same signal object.
 
         """
-        try:
-            return self[name]
-        except KeyError:
-            return self.setdefault(name, NamedSignal(name, doc))
+        return self.setdefault(name, NamedSignal(name, doc))
```

Comment:
I've seen this pattern quite a bit in the wild:

    signal('something-happened').send()

So this kind of creation can be quite wasteful.